src/main/java/com/social_media/social_media/repository/post/PostRepositoryImpl.java

```diff
@@ -53,5 +53,20 @@ public List<Post> findAll() {
         return postList;
     }
 
+    @Override
+    public List<Post> findSellerById(Long userIdToFollow) {
+        List<Post> filteredPosts = new ArrayList<>();
+        for (Post post : posts.values()) {
```

Comment:
No usar values(), usar el foreach directo sobre el mapa 
```suggestion
       posts.forEach((__, post) -> { 
          if (post.getUserId().equals(userIdToFollow)) {
                   filteredPosts.add(post);
           }
        });
```